tests/e2e/kserve_test.go

```diff
@@ -410,3 +413,203 @@ func (tc *KserveTestCtx) validateTemplatedResourceOwnerRefsAndLabels(expectOwned
 			WithCustomErrorMsg(msg, child.gvk.Kind, child.nn.Name, child.nn.Namespace))
 	}
 }
+
+func (tc *KserveTestCtx) ValidateCustomCertificateCreation(t *testing.T) {
+	t.Helper()
+
+	originalDSC := tc.FetchDataScienceCluster()
+
+	defer func() {
```

Comment:
@ugiordan just to continue with your previous conversation, this is probably the only way how to ensure the cluster is in the original state even if the test fails, right?

Reply:
Moved restore to the end of each test case.